dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java

```diff
@@ -0,0 +1,311 @@
+
+        datasetRecordDefinitions[0]= new FieldDef("RootRecord", FieldType.RECORD, "rec", 4, false, false, HpccSrcType.LITTLE_ENDIAN, fieldDefs);
+
+        // String key
+        fieldDefs = new FieldDef[2];
+        fieldDefs[0] = new FieldDef("key", FieldType.STRING, "STRING4", 4, true, false, HpccSrcType.SINGLE_BYTE_CHAR, new FieldDef[0]);
+        fieldDefs[1] = new FieldDef("payload", FieldType.STRING, "STRING16", 16, true, false, HpccSrcType.SINGLE_BYTE_CHAR, new FieldDef[0]);
+
+        datasetRecordDefinitions[1]= new FieldDef("RootRecord", FieldType.RECORD, "rec", 4, false, false, HpccSrcType.LITTLE_ENDIAN, fieldDefs);
+
+        for (int i = 0; i < datasetNames.length; i++)
+        {
+            //------------------------------------------------------------------------------
+            // Create indexable dataset
+            //------------------------------------------------------------------------------
+
+            ArrayList<HPCCRecord> partitionRangeStart = new ArrayList<HPCCRecord>();
+            ArrayList<HPCCRecord> partitionRangeEnd = new ArrayList<HPCCRecord>();
+
+            String datasetName = datasetNames[i];
+            FieldDef recordDef = datasetRecordDefinitions[i];
+            createIndexableFile(datasetName, recordDef, partitionRangeStart, partitionRangeEnd);
+            
+            //------------------------------------------------------------------------------
+            // Create index
+            //------------------------------------------------------------------------------
+
+            String indexName = datasetName + "::key";
+            String ecl = "rec := " + RecordDefinitionTranslator.toECLRecord(recordDef);
+            ecl += "ds := DATASET('" + datasetName + "', rec, THOR);";
+            ecl += "idx := INDEX(ds, {key}, {payload},'" + indexName + "');";
+            ecl += "BUILDINDEX(idx, OVERWRITE);";
+            
+            WorkunitWrapper wu = new WorkunitWrapper();
+            wu.setECL(ecl);
+            wu.setJobname("IndexCreation" + datasetName);
+            wu.setCluster(thorcluster);
+
+            try
+            {
+                HPCCWsWorkUnitsClient client = wsclient.getWsWorkunitsClient();
+                String result = client.createAndRunWUFromECLAndGetResults(wu);
+            }
+            catch(Exception e)
+            {
+                Assert.fail("Test failed with error: " + e.getMessage());
+            }
+
+            //------------------------------------------------------------------------------
+            // Read index and check TLK against known partition ranges
+            //------------------------------------------------------------------------------
+            
+            HPCCFile file = new HPCCFile(indexName, connString , hpccUser, hpccPass);
+
+            DataPartition[] fileParts = file.getFileParts();
+            PartitionProcessor processor = new PartitionProcessor(recordDef, fileParts);
```

Comment:
per offline conversation, let's consider moving PartitionProcessor within HPCCfile, and expose findMatchingPartitions(filter), also, let's make sure we document the filter range and it's use